docs-src/concepts/what-is-worker-versioning.md

```diff
@@ -0,0 +1,223 @@
+`2.0`, Temporal Server assumes that Workers with this version can process the existing Event
+Histories successfully.
+
+Continue with your normal development cycle, adding a `3.0` version.
+Nothing new here:
+
+| set 1         | set 2         | set 3 (default) |
+| ------------- | ------------- | --------------- |
+| 1.0 (default) | 2.0           | 3.0 (default)   |
+|               | 2.1 (default) |                 |
+
+Now imagine that version `3.0` doesn't have an explicit bug, but something about the business logic
+is less than ideal. You are okay with existing `3.0` Workflows running to completion, but you want new
+Workflows to use the old `2.x` branch. This operation is supported by performing an update targeting
+`2.1` (or `2.0`) and setting it's set as the current default, which results in these sets:
+
+| set 1         | set 3         | set 2 (default) |
+| ------------- | ------------- | --------------- |
+| 1.0 (default) | 3.0 (default) | 2.0             |
+|               |               | 2.1 (default)   |
+
+Now new Workflows start on `2.1`.
+
+#### Permitted and forbidden operations on version sets
+
+A request to change the sets can do one of the following:
+
+- Add a version to the sets as new the default version in a new overall-default compatible set.
+- Add a version to an existing set that's compatible with an existing version.
+  - Optionally making it the default for that set
+  - Optionally making that set the overall-default set.
+- Promote a version within an existing set to become the default for that set.
+- Promote a set to become the overall-default set.
+
+You can't explicitly delete versions. This helps you avoid the situation in which Workflows
+accidentally become stuck with no means of making progress because the version they're associated
+with no longer exists.
+
+However, sometimes you might want to do this intentionally. If you _want_ to make sure that all
+Workflows currently being processed by, say, `2.0` stop (even if you don't yet have a new version
+ready) you can add a new version `2.1` to the sets marked as compatible with `2.0`. New tasks will
+target `2.1`, but since you haven't deployed any `2.1` Workers, they won't make any
+progress.
+
+#### Set constraints
+
+The sets have a maximum size limit, which defaults to 1000 build ids across all sets. This limit is
+configurable on Temporal Server via the `limit.versionBuildIDsPerQueue` dynamic config property.
+Operations to add new Build IDs to the sets will fail if the limit would be exceeded.
+
+There is also a limit on the number of sets, which defaults to 10. This limit is configurable via
+the `limit.versionCompatibleSetsPerQueue` dynamic config property.
+
+In practice, these limits should rarely be a concern because a version is no longer needed after no
+open Workflows are using that version, and a background process will delete IDs and sets which are
+no longer needed.
```

Comment:
```suggestion
In practice, these limits should rarely be a concern because a version is no longer needed after no open Workflows are using that version, and a background process will delete IDs and sets that are no longer needed.
```